### PR TITLE
Docs generation is now `python -m buildconfig docs` and `python -m buildconfig docs full_generation`

### PR DESCRIPTION
**Pull request by oddbookworm**
I didn't want to have to invoke a specific python file every time I generate docs, so here's a useful helper

### CODE REVIEW

**Review by damusss — COMMENTED** [review]
When running it from setup.py it says `Please use the following replacement: D:\Programs\Python312\python.exe buildconfig docs full_generation`  but running that command raises `ImportError: attempted relative import with no known parent package` as the suggestion is missing '`-m`'. Other than that, I like this helper

**Comment by oddbookworm** [comment]
> When running it from setup.py it says `Please use the following replacement: D:\Programs\Python312\python.exe buildconfig docs full_generation` but running that command raises `ImportError: attempted relative import with no known parent package` as the suggestion is missing '`-m`'. Other than that, I like this helper

Oops, it's fixed now

**Review by damusss — APPROVED** [review]
LGTM, I also found it a bit annoying before

File: buildconfig/config.py [hunk 1]
```diff
@@ -165,6 +166,10 @@ def main(auto=False):
 Only SDL2 is supported now.""")
 
     kwds = {}
+
+    if "docs" in sys.argv:
```
Comment (ankith26) [inline buildconfig/config.py:170]:
```suggestion
    if sys.argv[1] == "docs":
```
I would like this to be more stricter

Comment (oddbookworm) [reply buildconfig/config.py]: Committed

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada: 

PS: the fails seem unrelated so I will merge this PR in now

**Pull request merged**